integration/src/bitcoin/bitcoin.ts

```diff
@@ -294,74 +281,74 @@ export function bitcoinTests(get: () => { wallet: HDWallet; info: HDWalletInfo }
-                  },
-                  txid: "c16a03f1cf8f99f6b5297ab614586cacec784c2d259af245909dedb0e39eddcf",
-                },
-                {
-                  vout: 1,
-                  valueSat: 200000,
-                  sequence: 4294967295,
-                  scriptSig: {
-                    hex:
-                      "48304502200fd63adc8f6cb34359dc6cca9e5458d7ea50376cbd0a74514880735e6d1b8a4c0221008b6ead7fe5fbdab7319d6dfede3a0bc8e2a7c5b5a9301636d1de4aa31a3ee9b101410486ad608470d796236b003635718dfc07c0cac0cfc3bfc3079e4f491b0426f0676e6643a39198e8e7bdaffb94f4b49ea21baa107ec2e237368872836073668214",
-                  },
-                  txid: "1ae39a2f8d59670c8fc61179148a8e61e039d0d9e8ab08610cb69b4a19453eaf",
-                },
-              ],
-              vout: [
-                {
-                  value: "0.00390000",
-                  scriptPubKey: {
-                    hex: "76a91424a56db43cf6f2b02e838ea493f95d8d6047423188ac",
-                  },
-                },
-              ],
-            },
+            tx,
             hex:
               "0100000002cfdd9ee3b0ed9d9045f29a252d4c78ecac6c5814b67a29b5f6998fcff1036ac1010000008b483045022072ba61305fe7cb542d142b8f3299a7b10f9ea61f6ffaab5dca8142601869d53c0221009a8027ed79eb3b9bc13577ac2853269323434558528c6b6a7e542be46e7e9a820141047a2d177c0f3626fc68c53610b0270fa6156181f46586c679ba6a88b34c6f4874686390b4d92e5769fbb89c8050b984f4ec0b257a0e5c4ff8bd3b035a51709503ffffffffaf3e45194a9bb60c6108abe8d9d039e0618e8a147911c68f0c67598d2f9ae31a010000008b48304502200fd63adc8f6cb34359dc6cca9e5458d7ea50376cbd0a74514880735e6d1b8a4c0221008b6ead7fe5fbdab7319d6dfede3a0bc8e2a7c5b5a9301636d1de4aa31a3ee9b101410486ad608470d796236b003635718dfc07c0cac0cfc3bfc3079e4f491b0426f0676e6643a39198e8e7bdaffb94f4b49ea21baa107ec2e237368872836073668214ffffffff0170f30500000000001976a91424a56db43cf6f2b02e838ea493f95d8d6047423188ac00000000",
           },
         ];
-        let outputs: BTCSignTxOutput[] = [
+        let outputs: core.BTCSignTxOutput[] = [
           {
             address: "bc1qksxqxurvejkndenuv0alqawpr3e4vtqkn246cu",
-            addressType: BTCOutputAddressType.Spend,
-            scriptType: BTCOutputScriptType.PayToAddress,
+            addressType: core.BTCOutputAddressType.Spend,
             amount: String(390000 - 10000),
             isChange: false,
           },
           {
-            address: "1FH6ehAd5ZFXCM1cLGzHxK1s4dGdq1JusM",
-            addressType: BTCOutputAddressType.Change,
-            scriptType: BTCOutputScriptType.PayToAddress,
+            addressNList: core.bip32ToAddressNList("m/44'/0'/0'/0/0"),
+            addressType: core.BTCOutputAddressType.Change,
+            scriptType: core.BTCOutputScriptType.PayToAddress,
             amount: String(9000),
-            isChange: false,
```

Comment:
Here's some evidence of the type confusion we've had going -- passing integration tests with `addressType: BTCOutputAddressType.Change` and `isChange: false`. (WTF?)

Reply:
in practice, this strangeness is an attempt to reconcile `hdwallet-keepkey`'s version of `validateVoutOrdering` -- which requires `BTCOutputAddressType.Change` in the second position -- with the usage of a literal change address in the test. This shouldn't work, because the whole point of `BTCOutputAddressType.Change` is to be used with an `addressNList` and `scriptType` for the wallet to calculate the address. (This is required so that your KK knows you own the address of the change output; otherwise it wouldn't be safe for it to subtract the change amount from the amount it has you confirm the spending of.)      _just so happens_ that validate